codegen-winter/src/air/transition_constraints.rs

```diff
@@ -40,18 +40,18 @@ pub(super) fn add_fn_evaluate_aux_transition(impl_ref: &mut Impl, ir: &AirIR) {
     // define the function.
     let evaluate_aux_transition = impl_ref
         .new_fn("evaluate_aux_transition")
-        .generic("E: FieldElement<BaseField = Felt> + ExtensionOf<F>")
-        .generic("F: FieldElement<BaseField = Felt>")
+        .generic("E: FieldElement<BaseField = Felt>")
+        .generic("F: FieldElement<BaseField = Felt> + ExtensionOf<E>")
```

Comment:
These are now backwards from what we usually do, so we should stick to opposite lettering that we had before (`F` for the base field and `E` for extension). I assume you made the change due to compilation errors, but maybe if you reverse the order of the generics the issue will be resolved?

```
        .generic("F: FieldElement<BaseField = Felt>")
        .generic("E: FieldElement<BaseField = Felt> + ExtensionOf<F>")
```

If possible, it would actually be better to have a "where" clause exactly like in Miden/Winterfell, if `codegen` supports that

Reply:
Fixed, thanks. Codegen does support where clauses.

Reply:
Nice, that's great